vernac/attributes.ml

```diff
@@ -15,9 +15,20 @@ type vernac_flags = vernac_flag list
 and vernac_flag = string * vernac_flag_value
 and vernac_flag_value =
   | VernacFlagEmpty
+  | VernacFlagBool of bool
```

Comment:
We already have `att(true)`, why add an equivalent `att=on`?

Reply:
You mean that the syntax `att(true)` is already supported, but we don't have any actual `true` sub-attribute anywhere in sight. Correct? My suggestion was indeed a generic sub-attribute `on` / `off` rather than a new syntax for taking booleans. But I am fine with both ways.

Reply:
Not sure what an "actual" sub attribute is, but program and canonical both take true/false that way.

Reply:
I think a typed version is clearer, the way `att(true)` is implemented seems a bit hacky in particular when we also support the `= ..` leaves. [The same could be said on `since="foo"` vs `since("foo")` right?

The syntax `typing(guarded(no))` seems a bit hard to read IMHO, but indeed, OMMV.

I guess I'd migrate the ones using the bool hack to the new syntax, but also I understand that this is totally a matter of personal preference.

Reply:
> Not sure what an "actual" sub attribute is, but program and canonical both take true/false that way.

Oh wow! Thanks for teaching me this, this is not documented anywhere, so I didn't realize.

Reply:
I tend to lean on the side of Emilio here. Besides, since `program(true/false)` and `canonical(true/false)` are not documented yet, it is still a good time to do this kind of change. I bet they are also used nowhere.

Reply:
Yes we can make that change indeed. After thinking a bit I support:

- `foo=bar` when `foo` can take a _single_ value
- `foo(bar1,bar2)` when indeed `foo` expects a list of arguments.

WDYT folks?

Reply:
Actually, I was not totally correct: `canonical(false)` is documented: https://coq.inria.fr/refman/language/extensions/canonical.html#coq:attr.canonical-false
So at least for this one, we will need a deprecation phase if it is changed to `canonical=off`.

Reply:
Before the style of attributes diverges too much between `foo(true)`, `foo=true`, `foo=on`, `foo="on`, `foo="true"`, etc. and someone complains that it is fully inconsistent, can't we fix a few rules? First choice, a dictator or a committee? If a dictactor, I propose to appoint @skyskimmer, who made the implementation, as "attributes" dictactor. If a committee, I'm ok to coordinate a group whose purpose is to deliver guidelines for attributes (but I'm ok that someone else coordinates too, as my ultimate priority is that we converge, whatever the way we converge). @ppedrot, @mattam82, @ejgallego, @zimmi48, @maximedenes, @gares ... are you ok with this proposal?